cmake/FindWdk.cmake

```diff
@@ -37,131 +37,131 @@ file(GLOB WDK_NTDDK_FILES
 
 if(WDK_NTDDK_FILES)
     list(GET WDK_NTDDK_FILES -1 WDK_LATEST_NTDDK_FILE)
-endif()
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(WDK REQUIRED_VARS WDK_LATEST_NTDDK_FILE)
-
-get_filename_component(WDK_ROOT ${WDK_LATEST_NTDDK_FILE} DIRECTORY)
-get_filename_component(WDK_ROOT ${WDK_ROOT} DIRECTORY)
-get_filename_component(WDK_VERSION ${WDK_ROOT} NAME)
-get_filename_component(WDK_ROOT ${WDK_ROOT} DIRECTORY)
-get_filename_component(WDK_ROOT ${WDK_ROOT} DIRECTORY)
-
-message(STATUS "WDK_ROOT: " ${WDK_ROOT})
-message(STATUS "WDK_VERSION: " ${WDK_VERSION})
-
-set(WDK_WINVER "0x0601" CACHE STRING "Default WINVER for WDK targets")
-
-set(WDK_ADDITIONAL_FLAGS_FILE "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/wdkflags.h")
-file(WRITE ${WDK_ADDITIONAL_FLAGS_FILE} "#pragma runtime_checks(\"suc\", off)")
-
-set(WDK_COMPILE_FLAGS
-    "/Zp8" # set struct alignment
-    "/GF"  # enable string pooling
-    "/GR-" # disable RTTI
-    "/Gz" # __stdcall by default
-    "/kernel"  # create kernel mode binary
-    "/FIwarning.h" # disable warnings in WDK headers
-    "/FI${WDK_ADDITIONAL_FLAGS_FILE}" # include file to disable RTC
-    )
-
-set(WDK_COMPILE_DEFINITIONS "WINNT=1")
-set(WDK_COMPILE_DEFINITIONS_DEBUG "MSC_NOOPT;DEPRECATE_DDK_FUNCTIONS=1;DBG=1")
-
-if(CMAKE_SIZEOF_VOID_P EQUAL 4)
-    list(APPEND WDK_COMPILE_DEFINITIONS "_X86_=1;i386=1;STD_CALL")
-    set(WDK_PLATFORM "x86")
-elseif(CMAKE_SIZEOF_VOID_P EQUAL 8)
-    list(APPEND WDK_COMPILE_DEFINITIONS "_WIN64;_AMD64_;AMD64")
-    set(WDK_PLATFORM "x64")
-else()
-    message(FATAL_ERROR "Unsupported architecture")
-endif()
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(WDK REQUIRED_VARS WDK_LATEST_NTDDK_FILE)
 
```

Comment:
I don't like putting almost the whole file into the `if` clause. It's better to use `return()` and put the check after `find_package_handle_standard_args`:
```cmake
if (NOT WDK_LATEST_NTDDK_FILE)
    return()
endif()
```

Reply:
It didn't make sense to check `WDK_NTDDK_FILES` and then, a few lines later, check `WDK_LATEST_NTDDK_FILE`, so I simply checked the former and returned if it is empty.

Also, the check can't be moved to after `find_package_handle_standard_args` because that's the function that's causing the "abort" (see my original request message).

I think my latest push addresses your concerns.

Reply:
Yeap, but `find_package_handle_standard_args` is crucial for proper work of `find_package`. You can use:
- `find_package(WDK REQUIRED)` - this will generate an error if no WDK is found
- `find_package(WDK)` - this will generate a warning if no WDK is found
- `find_package(WDK QUIET)` - this will generate no warnings nor errors if no WDK is found

So the `return()` should be after `find_package_handle_standard_args`. In your case you just need to use `find_package(WDK QUIET)`.

Reply:
Gotcha. Done.

(Thanks  for the tip on the `find_package(WDK)`. I like the warning, just don't want the failure.)